python_modules/libraries/dagster-sling/dagster_sling/components/common.py

```diff
@@ -0,0 +1,157 @@
+    AssetAttributesModel,
+    ComponentLoadContext,
+    ResolutionContext,
+    Resolvable,
+    Resolver,
+)
+from dagster.components.resolved.core_models import AssetPostProcessor, OpSpec
+from dagster.components.utils import TranslatorResolvingInfo
+from typing_extensions import TypeAlias
+
+from dagster_sling.asset_decorator import sling_assets
+from dagster_sling.dagster_sling_translator import DagsterSlingTranslator
+from dagster_sling.resources import AssetExecutionContext, SlingResource
+
+SlingMetadataAddons: TypeAlias = Literal["column_metadata", "row_count"]
+
+
+class DagsterSlingTranslatorAutomationCondition(DagsterSlingTranslator):
+    """Subclassed to pull in any overridden impl for get_automation_condition
+    into produced asset specs.
+    """
+
+    def get_automation_condition(
+        self, stream_definition: Mapping[str, Any]
+    ) -> Optional[AutomationCondition]: ...
+
+    def get_asset_spec(self, stream_definition: Mapping[str, Any]) -> AssetSpec:
+        asset_spec = super().get_asset_spec(stream_definition)
+        return asset_spec.replace_attributes(
+            automation_condition=self.get_automation_condition(stream_definition)
+        )
+
+
+class ComponentsDagsterSlingTranslator(DagsterSlingTranslator):
+    def __init__(self, *, resolving_info: TranslatorResolvingInfo):
+        super().__init__()
+        self.resolving_info = resolving_info
+
+    def get_asset_spec(self, obj: Any) -> AssetSpec:  # pyright: ignore[reportIncompatibleMethodOverride]
+        base_spec = super().get_asset_spec(obj)
+        return self.resolving_info.get_asset_spec(
+            base_spec,
+            {self.resolving_info.obj_name: obj, "spec": base_spec},
+        )
+
+
+def resolve_translator(
+    context: ResolutionContext,
+    asset_attributes,
+) -> DagsterSlingTranslator:
+    # TODO: Consider supporting owners and code_version in the future
+    if asset_attributes and isinstance(asset_attributes, AssetAttributesModel):
+        set_vals = asset_attributes.model_dump(exclude_unset=True)
+        if "owners" in set_vals:
+            raise ValueError("owners are not supported for sling_replication_collection component")
+        if "code_version" in set_vals:
+            raise ValueError(
+                "code_version is not supported for sling_replication_collection component"
+            )
+
+    return ComponentsDagsterSlingTranslator(
+        resolving_info=TranslatorResolvingInfo(
+            "stream_definition",
+            asset_attributes or AssetAttributesModel(),
+            context,
+        )
+    )
+
+
+@dataclass
+class SlingReplicationSpecModel(Resolvable):
+    path: str
+    op: Optional[OpSpec] = None
+    translator: Annotated[
+        Optional[DagsterSlingTranslator],
+        Resolver(
+            resolve_translator,
+            model_field_name="asset_attributes",
+            model_field_type=Union[str, AssetAttributesModel],
+        ),
+    ] = None
```

Comment:
should we move to the `translation` pattern while we are mucking with this?

Reply:
yeah for sure will do that in followup